projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -516,7 +517,11 @@ export class IgxDatePickerComponent extends PickerBaseDirective implements Contr
         private cdr: ChangeDetectorRef,
         @Optional() @Inject(IGX_INPUT_GROUP_TYPE) _inputGroupType?: IgxInputGroupType) {
         super(element, _localeId, _inputGroupType);
-        this.locale = this.locale || this._localeId;
+        initi18n(_localeId);
+        this.locale = this.locale || getCurrentI18n();
+        getI18nManager().onResourceChange(() => {
+            this._resourceStrings = getCurrentResourceStrings(DatePickerResourceStringsEN, false);
```

Comment:
Shouldn't the locale be updated on change, like in the grid?:
```
        getI18nManager().onResourceChange((args: ResourceChangeEventArgs) => {
            this._defaultLocale = args.newLocale;
```

Because currently the picker does not react to a change in the locale of the page and stays in the original language/formattings.

It works only if you set it before the picker initializes.